supply/spec/uploader_spec.rb

```diff
@@ -203,22 +203,69 @@ def find_obbs
       end
     end
 
-    describe '#perform_upload' do
-      let(:client) { double('client') }
-      let(:config) { { apk: 'some/path/app.apk' } }
+    RSpec::Matchers.define(:same_localizedtext_as) do |expected_arr|
+      match do |actual_arr|
+        actual_arr.sort_by(&:language).zip(expected_arr.sort_by(&:language)).map do |actual_localization, expected_localization|
+          actual_localization.language == expected_localization.language && actual_localization.text == expected_localization.text
+        end.reduce(:&)
```

Comment:
Shouldn't that be `reduce(:&&)` to join all tests by a logical AND operator, as opposed to use a **bitwise**-and `&` operator?
Though, in practice, if that's really the intent, you'd be better served by using `all_satisfy?` method instead.

Also, I wonder if using the built-in `match_array` RSpec matcher (which compares content of arrays without regard for the potential different order of the elements being compared) wouldn't be enough to compare those two arrays, instead of writing your custom matcher? (Though I'm not 100% if `AndroidPublisher::LocalizedText` has a proper implementation of `==` that compares each property for equality instead of using ObjectIDs…)

Reply:
I used `reduce(:&)` because `&` is the operator we have for [TrueClass](https://ruby-doc.org/2.7.8/TrueClass.html) and [FalseClass](https://ruby-doc.org/2.7.8/FalseClass.html) (there's no `&&`). I would prefer using the expression in that form rather than expand it to `reduce { |a, b| a && b }`. What do you think?

---

As for the matcher, I actually had to write a new one because we lack the support for `==` on `AndroidPublisher::LocalizedText` 😕 , so even `match_array` wouldn't help in that case. (it does a plain comparison of object ids as you implied last).

Reply:
> I used `reduce(:&)` because `&` is the operator we have for [TrueClass](https://ruby-doc.org/2.7.8/TrueClass.html) and [FalseClass](https://ruby-doc.org/2.7.8/FalseClass.html) (there's no `&&`). I would prefer using the expression in that form rather than expand it to `reduce { |a, b| a && b }`. What do you think?

Then in that case I'd rather have you use `all_satisfy?`, which is more idiomatic Ruby for that kind of thing

> As for the matcher, I actually had to write a new one because we lack the support for `==` on `AndroidPublisher::LocalizedText` 😕 , so even `match_array` wouldn't help in that case. (it does a plain comparison of object ids as you implied last).

What about declaring that `==` ourselves then? Since Ruby allows to re-open classes to add definitions 🙃 

```ruby
class AndroidPublisher::LocalizedText
  def ==(other)
    self.language == other.language && self.text == other.text
  end
end
```

Reply:
Besides, I think your implementation of that matcher has a hidden bug, because in `a.zip(b)`, if `a` has less elements than `b` then the result of the `zip` will only have as many elements as `a`, ignoring the trailing elements of `b`:
```ruby
$ pry
[1] pry(main)> a = [1,2]
=> [1, 2]
[2] pry(main)> b = [3,4,5]
=> [3, 4, 5]
[3] pry(main)> a.zip(b)
=> [[1, 3], [2, 4]]
```

Which means that if `actual_arr` has less elements than `expected_arr`, your matcher will pass instead of detecting a mismatch. Which is a bug that I don't thing `match_array` is subject to 🙃

Reply:
The `.zip` "bug" is actually a good point. The code should be changed.

---

I forgot about that we can open classes anywhere we want, that was a good call 😄 . I added the `==` override as you suggested and that simplified the test code a lot. Now we can _get away_ with just a `match_all` call. That should solve the `zip` problem too.